### PR TITLE
make Version.version a VersionString, and VersionString a complete object

### LINKED ISSUE #15798: make Version.version (and other version fields) a type of VersionString
We recently refactored version string comparison to fix some edge cases but it would be good if Version.version was naturally able to be compared as a version string everywhere.  This will lead to more straightforward code where version is compared to, and will remove any edge case where we accidentally forget to convert it first

### PR DESCRIPTION
**Pull request by eviljeff**
fixes  #15798 - supersedes #15800

### CODE REVIEW

File: src/olympia/blocklist/admin.py [hunk 1]
```diff
@@ -423,6 +423,7 @@ def add_view(self, request, **kwargs):
             'save_as': False,
             'block_history': self.block_history(
                 self.model(input_guids=guids_data)),
+            'submission_complete': False,
```
Comment (bobsilverberg) [inline src/olympia/blocklist/admin.py:426]: How is this change related to the issue?

File: src/olympia/versions/compare.py [hunk 2]
```diff
@@ -99,22 +108,22 @@ def __gt__(self, other):
...
 
 
 def version_int(version):
     """This is used for converting an app version's version string into a
     single number for comparison.  To maintain compatibility the minor parts
     are limited to 99 making it unsuitable for comparing addon version strings.
     """
-    vstr = VersionString(version)
-    vdict = dict(vstr.full_dict)
+    vdict = dict(VersionString(version).parts)
```
Comment (bobsilverberg) [inline src/olympia/versions/compare.py:126]: I might be missing something here, but isn't this patch about making `version` always be a `VersionString`? Why do we need `VersionString(version)`?

**Review by bobsilverberg — APPROVED** [review]
Looks good, thanks @eviljeff. I have a couple of questions.

r+wc

File: src/olympia/blocklist/admin.py (hunk 1 above)
Comment (eviljeff) [reply src/olympia/blocklist/admin.py:426]: it was a drive-by fix for a warning about undefined template variables.

File: src/olympia/versions/compare.py (hunk 2 above)
Comment (eviljeff) [reply src/olympia/versions/compare.py:126]: The changes are related to add-on versions, but `version_int` is used by _application_ version comparison code - that doesn't use `VersionString` (yet).

File: src/olympia/blocklist/admin.py (hunk 1 above)
Comment (eviljeff) [reply src/olympia/blocklist/admin.py:426]: I could remove it again.

Comment (bobsilverberg) [reply src/olympia/blocklist/admin.py:426]: No need to do that. I was just curious.

**Review by diox — APPROVED** [review]
